### PR TITLE
Update versions of external packages we reference

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
WIP so that we can reference the new All-in-one-search APIs directly.

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -172,11 +172,12 @@
     <MicrosoftVisualStudioProjectSystemVersion>17.0.77-pre-g62a6cb5699</MicrosoftVisualStudioProjectSystemVersion>
     <MicrosoftVisualStudioRemoteControlVersion>16.3.44</MicrosoftVisualStudioRemoteControlVersion>
     <MicrosoftVisualStudioSDKAnalyzersVersion>16.10.10</MicrosoftVisualStudioSDKAnalyzersVersion>
+    <MicrosoftVisualStudioSearchVersion>17.4.0-preview-1-32724-041</MicrosoftVisualStudioSearchVersion>
```
Comment (CyrusNajmabadi) [inline eng/Versions.props:175]: This is where the new APIs live.

**Comment by JoeRobich** [comment]
If these packages shipped in 17.4 P1 we may get away with not re-creating main-vs-deps, since 17.4 P1 images are available for integration CI.

**Comment by CyrusNajmabadi** [comment]
> If these packages shipped in 17.4 P1 

They did not unfortunately.  Some only shipped yesterday, and some more are still coming.

**Comment by JoeRobich** [comment]
I believe @chsienki was going to need to update S.C.I to 6.0.0 for us to release 17.3 packages. Perhaps we can pull in those changes when they are available.

**Comment by CyrusNajmabadi** [comment]
Sounds good.  Thanks!

**Comment by CyrusNajmabadi** [comment]
@JoeRobich @tmat @jasonmalinowski ptal.  Thanks!

File: src/Dependencies/CodeAnalysis.Debugging/Microsoft.CodeAnalysis.Debugging.Package.csproj [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>netstandard1.3;net45</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0</TargetFrameworks>
```
Comment (tmat) [inline src/Dependencies/CodeAnalysis.Debugging/Microsoft.CodeAnalysis.Debugging.Package.csproj:5]: `TargetFramework`

Comment (CyrusNajmabadi) [inline src/Dependencies/CodeAnalysis.Debugging/Microsoft.CodeAnalysis.Debugging.Package.csproj:5]:
```suggestion
    <TargetFramework>netstandard2.0</TargetFramework>
```

File: src/Dependencies/PooledObjects/Microsoft.CodeAnalysis.PooledObjects.Package.csproj [hunk 3]
```diff
@@ -2,7 +2,7 @@
 <!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>netstandard1.3;net45</TargetFrameworks>
+    <TargetFrameworks>netstandard2.0</TargetFrameworks>
```
Comment (CyrusNajmabadi) [inline src/Dependencies/PooledObjects/Microsoft.CodeAnalysis.PooledObjects.Package.csproj:5]:
```suggestion
    <TargetFramework>netstandard2.0</TargetFramework>
```

File: eng/Versions.props [hunk 4]
```diff
@@ -283,12 +284,12 @@
...
     -->
     <NewtonsoftJsonVersion>13.0.1</NewtonsoftJsonVersion>
-    <StreamJsonRpcVersion>2.12.7-alpha</StreamJsonRpcVersion>
+    <StreamJsonRpcVersion>2.12.25</StreamJsonRpcVersion>
     <!--
       When updating the S.C.I or S.R.M version please let the MSBuild team know in advance so they
       can update to the same version. Version changes require a VS test insertion for validation.
     -->
-    <SystemCollectionsImmutableVersion>5.0.0</SystemCollectionsImmutableVersion>
+    <SystemCollectionsImmutableVersion>6.0.0</SystemCollectionsImmutableVersion>
```
Comment (JoeRobich) [inline eng/Versions.props:292]: MSBuild and VS are both on 6.0.0 S.C.I so we are likely good here.

Comment (JoeRobich) [reply eng/Versions.props:292]: They are also on 6.0.0 S.R.M in case that is a change we want to make at this time.

**Comment by JoeRobich** [comment]
Changes look good to me. I think we should do a PR Validation prior to merging.  But need to wait for the current outage to be mitigated.

**Comment by CyrusNajmabadi** [comment]
> I think we should do a PR Validation prior to merging. 

How do we do that?

**Comment by JoeRobich** [comment]
PR Val: https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/419768 (microsoft)

File: eng/Versions.props [hunk 5]
```diff
@@ -25,24 +25,24 @@
     <MicrosoftVisualStudioExtensibilityTestingVersion>0.1.132-beta</MicrosoftVisualStudioExtensibilityTestingVersion>
     <!-- CodeStyleAnalyzerVersion should we updated together with version of dotnet-format in dotnet-tools.json -->
     <CodeStyleAnalyzerVersion>4.3.0-1.final</CodeStyleAnalyzerVersion>
-    <VisualStudioEditorPackagesVersion>17.3.133-preview</VisualStudioEditorPackagesVersion>
+    <VisualStudioEditorPackagesVersion>17.4.136-preview</VisualStudioEditorPackagesVersion>
     <!-- This should generally be set to $(VisualStudioEditorPackagesVersion),
          but sometimes EditorFeatures.Cocoa specifically requires a newer editor build. -->
```
Comment (jasonmalinowski) [inline eng/Versions.props:30]: The comment here seems a bit odd now that we're actually moving the Windows one ahead of the Mac one. Change it?

File: src/Compilers/Core/Portable/Compilation/SymbolInfo.cs [hunk 6]
```diff
@@ -14,7 +14,7 @@ public struct SymbolInfo : IEquatable<SymbolInfo>
     {
         internal static readonly SymbolInfo None = new SymbolInfo(null, ImmutableArray<ISymbol>.Empty, CandidateReason.None);
 
-        private ImmutableArray<ISymbol> _candidateSymbols;
+        private readonly ImmutableArray<ISymbol> _candidateSymbols;
```
Comment (jasonmalinowski) [inline src/Compilers/Core/Portable/Compilation/SymbolInfo.cs:17]: Why the readonly changes here? Did some analyzer get updated and is flagging this now?

**Review by jasonmalinowski — APPROVED** [review]

File: src/Compilers/Core/Portable/Compilation/SymbolInfo.cs (hunk 6 above)
Comment (CyrusNajmabadi) [reply src/Compilers/Core/Portable/Compilation/SymbolInfo.cs:17]: Yes. :)

**Pull request merged**